### DISCUSSION TITLE
Monitoring Frigate with Zabbix

### DISCUSSION THREAD

**Discussion by leccelecce** [blakeblackshear/frigate#9066] [Show and tell]
I use Zabbix at home for power/network/server/IP camera/VM/docker/application monitoring/dashboards/alerting. For those not familiar with it, best look here: https://www.zabbix.com/documentation/current/en/manual/introduction/features. This is about application/hardware monitoring, _not_ monitoring the actual video or motion/object events from Frigate.

I've configured an HTTP template for use with Frigate 0.13 (tested against beta7) which I intend to submit to the Zabbix community templates site. Dynamic discovery rules have been set up to auto-detect cameras, storage paths, detectors and GPUs so there is no hard-coding - it should work on anyone's setup. It uses Frigate's `/api/stats` HTTP endpoint, so just needs your Frigate hostname and the port if you're not using the default of 5000. There might be a few items which are not monitored e.g. I have an Intel GPU, but not NVIDIA, so the GPU stats I'm using are a bit more basic. I'll probably take a look at the stats code to see how non-Intel bits are structured at some point and update.

I know the Home Assistant add-on allows you to do much the same (if not more, because it subscribes to a broader range of MQTT topics), which is no doubt fine for some people. But if you're monitoring a reasonable amount of non-smart home hardware, you're probably not doing that in HASS, and I certainly prefer to have everything centralised.

The YAML is here which you can easily import into Zabbix UI.
https://gist.github.com/leccelecce/01cb0adeb57a21bc232c0067bb7b9d3c

In terms of what this looks like, below is what I've set up recently for three test cameras. Note that two components (hardware load average and disk response times) come not from Frigate but from the standard Zabbix Linux agent.

![image](https://github.com/blakeblackshear/frigate/assets/24962424/38e0717c-a18b-4d81-9d4c-dfc8a79f3948)

Raw data items, auto-discovered:

![image](https://github.com/blakeblackshear/frigate/assets/24962424/ade879c7-c0d4-4ae0-8814-d59df8f721c4)

Zabbix also allows configurable triggers for problem alerting. A few standard ones are included. I'm looking at setting some additional ones to monitor (e.g. unexpected zero FPS from a camera), but I need to play around with the stats API a bit more and might need to send PRs to add the odd bit of data (or add in some other API calls).

![image](https://github.com/blakeblackshear/frigate/assets/24962424/508c5dcc-80e1-48bb-8cff-cad58ec021cc)

If anyone else is using Zabbix 6.4+ and tries it out, all feedback welcome.

**Comment by phlundblom** [comment]
Cool 😎

**Comment by romuloreuwsaat** [comment]
In Zabbix 6.0 adjusting template, but the discovery list / Camera Discovery dont work...
I have que error on running the Discovery:
Preprocessing failed for: {"MyCamera_name":{"camera_fps":5.0,"capture_pid":448,"detection_enabled":1,"detection_fps":0.0,"f...
1. Failed: cannot extract value from json by path "$.cameras[*]~": cannot extract name from empty result

**Comment by DannyElfman34** [comment]
Just found out about that. Might give it a try (but I'm using Zabbix V7.
Does anyone manage to monitor Frigate logs ? I'm having trouble with some warning disconnecting the cameras and I would like to trap the message and get alerts from Zabbix . But I do not know where to start :(

**Reply by MarekWo** [reply]
I am using Zabbix 7.4 and I would like to give it a try too.
Did you manage to set this up on your V7?

**Reply by leccelecce** [reply]
I would have a look at https://www.zabbix.com/documentation/current/en/manual/config/items/itemtypes/log_items

**Comment by leccelecce** [comment]
I've re-exported this from Zabbix 7.2 and updated the gist above.

The updated version also adds a call to `/api/recordings/storage` to enable showing how much storage each camera is using.

Log monitoring is something I can have a look at. But I think you can just look at FPS counters being zero for that.

**Reply by MarekWo** [reply]
I confirm it does import perfectly in 7.4 and the data is now being collected. Great job!

**Reply by kkondyl** [reply]
Excellent work @leccelecce! I have imported the template to zabbix 7.0 which is LTS. I needed to manually change the version in the template. Moreover, I have added few more items in the template like CPU, RAM and TPU temperature. Tested with frigate 0.16RC1

**Reply by kayaman78** [reply]
> Excellent work @leccelecce! I have imported the template to zabbix 7.0 which is LTS. I needed to manually change the version in the template. Moreover, I have added few more items in the template like CPU, RAM and TPU temperature. Tested with frigate 0.16RC1

Please share!!!

Thanks for the great template leccelecce

**Reply by kkondyl** [reply]
@kayaman78 @leccelecce you can find the modified template here: https://gist.github.com/kkondyl/2f763b1102ee2a1676b4c273bd47d9a5

**Reply by kayaman78** [reply]
> @kayaman78 @leccelecce you can find the modified template here: https://gist.github.com/kkondyl/2f763b1102ee2a1676b4c273bd47d9a5

Very cool mod for discovery and coral, thanks!!

**Comment by leaskovski** [comment]
@leccelecce have you not been tempted to put this into a full git repo and summit it to the zabbix templates repo? https://www.zabbix.com/integrations
